index.js

```diff
@@ -1,11 +1,45 @@
 const express = require('express');
 const bodyParser = require('body-parser');
 const { bookRoutes, errorRoutes, customerRoutes } = require('./routers');
+const { bookDB, cartDB } = require('./db');
 const app = express();
 const port = 8080;
 
 app.use(bodyParser.json());
 
+// user wants to buy a book
+app.post('/users/:userId/cart/:cartId', (req, res) => {
+  const { userId, cartId } = req.params;
+  const booklist = req.body;
+
+  // check if the books are in the database
+
+  let bookIndexes = [];
+  for (let bookTitle of booklist) {
+    let titleIsFound = bookDB.some((bookEntry) => {
+      return bookEntry.title === bookTitle;
```

Comment:
in case u have pretty simple arrow function , then shorthand can be used `bookDB.some(({ title }) => title === bookTitle)`

Reply:
Nice catch!